### PR TITLE
L2-484: Improve Merge neurons UX

### PR DESCRIPTION
**Pull request by lmuntanerDfinity**
# Motivation

Improve the UX when merging neurons by disabling neurons if they can't be merged with the currently selected.

# Changes

* Rename SelectNeurons to SelectNeuronsToMerge.
* Move logic of allowing merging or not to util `mapMergeableNeurons`.

# Tests

* More tests for neuron util "mapMergeableNeurons".

### CODE REVIEW

**Comment by lmuntanerDfinity** [comment]
The BigO complexity has increased since now once the user selects a neuron, we need to check it against all the the others as well. But I tried having more than 20 neurons and the performance was perfect.

Here is a video of the new UX:

https://user-images.githubusercontent.com/98735375/165914255-c38f50bd-7933-4cc5-99e2-fa9390895ab7.mov

**Comment by lmuntanerDfinity** [comment]
@mstrasinskis @peterpeterparker please review

File: frontend/svelte/src/tests/lib/utils/neuron.utils.spec.ts [hunk 1]
```diff
@@ -926,6 +1067,32 @@ describe("neuron-utils", () => {
       };
       expect(canBeMerged([neuron, neuron2]).isValid).toBe(true);
     });
+
+    it("return invalid if one neurons have same followees on Manage Neuron and the other none", () => {
```
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/svelte/src/tests/lib/utils/neuron.utils.spec.ts:1075]: Thanks!

File: frontend/svelte/src/lib/utils/neuron.utils.ts [hunk 2]
```diff
@@ -346,7 +393,11 @@ const sameManageNeuronFollowees = (neurons: NeuronInfo[]): boolean => {
   if (sortedFollowees.length === 0) {
     return true;
   }
-  // Check that all neurons have same followees
+  // If values of list of followees is different than neurons.
```
[the remark this answers is not in the document]
Comment (lmuntanerDfinity) [reply frontend/svelte/src/lib/utils/neuron.utils.ts]: changed!

**Review by mstrasinskis — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**